cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -369,15 +369,15 @@ func isInstanceNotRunningYet(gceInstance *gce.ManagedInstance) bool {
 	return gceInstance.InstanceStatus == "" || gceInstance.InstanceStatus == "PROVISIONING" || gceInstance.InstanceStatus == "STAGING"
 }
 
-func generateInstanceName(migRef GceRef, existingNames map[string]bool) string {
+func generateInstanceName(baseName string, existingNames map[string]bool) string {
 	for i := 0; i < 100; i++ {
-		name := fmt.Sprintf("%v-%v", strings.TrimSuffix(migRef.Name, instanceGroupNameSuffix), rand.String(4))
+		name := fmt.Sprintf("%v-%v", strings.TrimSuffix(baseName, instanceGroupNameSuffix), rand.String(4))
```

Comment:
I guess we no longer need to trim suffix? I believe it could event produce incorrect behaviour if base name contains `instanceGroupNameSuffix`